src/Visitor/ReflectionVisitor.php

```diff
@@ -12,21 +12,32 @@
 use PhpParser\Node;
 use PhpParser\NodeVisitorAbstract;
 
-class WrappedFunctionInfoCollectorVisitor extends NodeVisitorAbstract
+class ReflectionVisitor extends NodeVisitorAbstract
 {
+    const REFLECTION_CLASS_KEY = 'reflectionClass';
     const IS_INSIDE_FUNCTION_KEY = 'isInsideFunction';
     const IS_ON_FUNCTION_SIGNATURE = 'isOnFunctionSignature';
     const FUNCTION_SCOPE_KEY = 'functionScope';
 
     private $scopeStack = [];
 
+    /**
+     * @var \ReflectionClass
+     */
+    private $reflectionClass;
+
     public function beforeTraverse(array $nodes)
     {
         $this->scopeStack = [];
+        $this->reflectionClass = null;
     }
 
     public function enterNode(Node $node)
     {
+        if ($node instanceof Node\Stmt\ClassLike) {
```

Comment:
What happens here if we declare an anonymous class within our class?
Will all the nodes after that still be seen as instances of that anonymous class?

Reply:
It crashes on anonymous classes in my test cases.
This could be fixed by

```php
public function enterNode(Node $node)
    {
        if ($node instanceof Stmt\Namespace_) {
            $this->namespace = $node->name;
        } elseif ($node instanceof Stmt\ClassLike) {
            if (null !== $node->name) {
                $node->fullyQualifiedClassName = Name::concat($this->namespace, $node->name);
            } else {
                $node->fullyQualifiedClassName = null;
            }
        }
    }
```
For the `FullyQualifiedClassNameVisitor`

and 
```php
    
public function enterNode(Node $node)
    {
        if ($node instanceof Node\Stmt\ClassLike && $node->fullyQualifiedClassName !== null) {
            $this->reflectionClass = new \ReflectionClass($node->fullyQualifiedClassName->toString());
        }
```

in the `ReflectionVisitor`

Reply:
Thank you @BackEndTea, I've added a test case.